### ISSUE TITLE
The "sort-imports" rule does not handle module-specifier-only imports.

### ISSUE THREAD

**Issue by theengineear** [eslint/eslint#12393]
**Tell us about your environment**

```
Environment Info:

Node version: v12.0.0
npm version: v6.9.0
Local ESLint version: v6.5.1 (Currently used)
Global ESLint version: Not found
```

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
// We should periodically check in on the "recommended" rules we're extending
// and remove redundancies:
// https://github.com/eslint/eslint/blob/master/conf/eslint-recommended.js
module.exports = {
  root: true,
  extends: 'eslint:recommended',
  rules: {
    eqeqeq: 'error',
    'comma-dangle': ['warn', 'always-multiline'],
    'no-console': 'warn',
    'no-prototype-builtins': 'warn',
    'no-shadow': 'warn',
    'no-undef-init': 'error',
    'no-unused-vars': 'warn',
    'no-var': 'error',
    'prefer-arrow-callback': ['error', { allowNamedFunctions: true }],
    'prefer-const': 'error',
    'quotes': ['warn', 'single', { avoidEscape: true, allowTemplateLiterals: true }],
    'semi': 'warn',
    'sort-imports': 'warn',
  },
  env: {
    es6: true,
    browser: true,
  },
  parserOptions: {
    ecmaVersion: 2020,
    sourceType: 'module',
  },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import '../b.js';
import 'https://example.com/elements/last.js';
import '../a.js';
```

```bash
eslint
```

**What did you expect to happen?**

I expected `eslint` to _warn_ me that my imports were not alphabetized. Ideally, it would complain until my imports were ordered as follows:

```js
import '../a.js';
import '../b.js';
import 'https://example.com/elements/last.js';
```

**What actually happened? Please include the actual, raw output from ESLint.**

There were _no warnings_ from `eslint`.

**Are you willing to submit a pull request to fix this bug?**

I'm unfamiliar with this codebase -- but potentially.

**Comment by g-plane** [comment]
Thanks for this report.

This is an expected behavior. The `sort-imports` rule doesn't check the import source.

**Comment by theengineear** [comment]
@g-plane , I'm surprised that this is the expected behavior. Should I open up a _change_ ticket?

**Comment by theengineear** [comment]
At a minimum, maybe an additional line could be added to the expected behavior block:

>Examples of correct code for this rule when using default options:

```js
/*eslint sort-imports: "error"*/
import 'module-without-export.js';
import * as bar from 'bar.js';
import * as foo from 'foo.js';
import {alpha, beta} from 'alpha.js';
import {delta, gamma} from 'delta.js';
import a from 'baz.js';
import b from 'qux.js';

/*eslint sort-imports: "error"*/
import 'ccc.js';
import 'bbb.js';
import 'aaa.js';

/*eslint sort-imports: "error"*/
import a from 'foo.js';
import b from 'bar.js';
import c from 'baz.js';

/*eslint sort-imports: "error"*/
import 'foo.js'
import * as bar from 'bar.js';
import {a, b} from 'baz.js';
import c from 'qux.js';

/*eslint sort-imports: "error"*/
import {a, b, c} from 'foo.js'
```

**Comment by g-plane** [comment]
Actually, different imports order may cause different behaviors. For example,

```js
// a.js
console.log('a')

export default 'a'
```

```js
// b.js
console.log('b')

export default 'b'
```

If we import `a.js` before `b.js` like this:

```js
import './a.js'
import './b.js'
```

Console will output:

```
a
b
```

However,

```js
import './b.js'
import './a.js'
```

Console will output:

```
b
a
```

**Comment by theengineear** [comment]
>Actually, different imports order may cause different behaviors...

Yep. That's of course true and why you wouldn't want to `--fix` this. I'm not refuting any of that. However, it just seems like `eslint` is being inconsistent here. I.e.,

NOT OK
```js
import b from './b.js';
import a from './a.js';
```

OK
```js
import './b.js';
import './a.js';
```

**Comment by g-plane** [comment]
That's a limitation of static analysis, and ESLint isn't able to know whether a module has side effects or not.

**Comment by theengineear** [comment]
>That's a limitation of static analysis, and ESLint isn't able to know whether a module has side effects or not.

OK, so if you can't know if there are side effects, why do you correct the following lines of code?

```js
import b from './b.js';
import a from './a.js';
```

Just because you have a named import doesn't mean the module you're importing from doesn't have side effects, right?

The point of sorting your imports is to decrease conflicts in version control. I don't understand why `eslint` would not sort my imports when I've specifically turned on the `sort-imports` rule. If I have to ensure some specific ordering of imports I'd disable the rule in the file.

**Comment by theengineear** [comment]
Again. I asked `eslint` to make sure my imports are sorted. I'd expect `eslint` to tell me when they're not.

**Comment by kaicataldo** [comment]
I agree that this is currently the intended behavior. Do you mind filling out the rule change template as described [here](https://eslint.org/docs/developer-guide/contributing/rule-changes)? You can feel free to add it in a comment in this issue here if you'd like.

Some question/comments I have:

1. We would probably need to put this behind an option as enabling it by default would lead to more warning.
2. How would these be sorted? By the path names? What rules would apply when sorting them? This is less clear than if the exports are named. I also want to note that, if side effects are an issue, that modules with exported bindings can be renamed to work around this, while an 'eslint-disable` comment would be required to turn this rule off for any imports that perform side effects where the order does matter.
